client/core/core.go

```diff
@@ -1899,6 +1899,67 @@ func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg
 		return err
 	}
 
+	ownsAddr := func(addr string) error {
+		owns, err := wallet.OwnsAddress(addr)
+		if err != nil {
+			return err
+		}
+		if !owns {
+			return fmt.Errorf("new wallet does not own address found in active trades: %v", addr)
+		}
+		return nil
+	}
+
+	// If there are active trades, make sure they can be settled by the
+	// keys held within the new wallet.
+	sameWallet := func() error {
+		for _, dc := range c.dexConnections() {
+			for _, trade := range dc.trackedTrades() {
+				if trade.isActive() {
+					t := trade.Trade()
+					waID := wallet.AssetID
+					// If the to asset, check if we own an
+					// contract.Address.
+					if (t.Sell && trade.Quote() == waID) ||
+						(!t.Sell && trade.Base() == waID) {
+						if err := ownsAddr(t.Address); err != nil {
+							return err
+						}
+						// Assume all trade addresses are
+						// owned by the new wallet if one is.
+						return nil
+					}
+					// If the from asset, check if we own a
+					// refund address for a match if any exist.
+					if trade.fromAssetID == waID {
+						for _, match := range trade.matches {
+							script := match.MetaData.Proof.Script
+							if len(script) == 0 {
+								continue
+							}
```

Comment:
What if the new wallet doesn't have the balance to cover the trades? What about locked outpoints?

This has the potential to get really messy, I think. I'm not sure that we can reject changes under any circumstances without creating the potential for wedged orders. There are some changes that simply cannot be made with active orders like changing the account, and some that absolutely must be allowed if we are to completely avoid wedged orders, like changing the wallet password.

Reply:
Maybe once wallets are internalized, this problem goes away? Will that bee real soon? Can consider this a bandaid?

Reply:
Even when we have built-ins, remote wallets will still be an option. 

We can probably get away with a simpler filter for now and punt on a comprehensive solution. But maybe just reject any changes when the wallet is the `fromWallet` on an active match. Don't even check the refund address, just error.

Reply:
> What if the new wallet doesn't have the balance to cover the trades? What about locked outpoints?

The purpose is to not allow changing to a different wallet with active trades. I think there is a change of the "toAsset" falling through if not yet matched because we don't have a refund address yet. Will the trade error out later if not enough funds?

>  changing the account

I see, we really need to check that an address belongs to an account, and not a wallet, to ensure the account hasn't changed?

> changing the wallet password.

There shouldn't be a scenario where this blocks that.

Should I work on something more complex, or can we is alright to specify this as temporary? Still guessing not an issue once wallets are internal by default.

Reply:
Actually, ownsAddress does check if it's this account that owns it.

Reply:
> when the wallet is the fromWallet on an active match.

The from or the to wallet? It looks like the refund address comes from the to wallet?

Is it alright to check, then error if we didn't find a refund address?

Reply:
We can maybe roll with the refund address check as you have it. Better to be over-persmissive, I guess. But note that the following situation is problematic. 

Our order matches as taker. Counterparty has broadcast. We have received `audit` request, but are waiting for confirmation before broadcasting our swap. In this case, if the user tries to reconfigure the `fromWallet`, even changing the account or wallet altogether, they will encounter no resistance, but will definitely error at swap time.

I wonder if we can reject changes when matches are specifically in `MakerSwapCast`?

Reply:
Still confused with to/from, but we are talking about having no refund address yet here?

Reply:
You have it right in the code.You only check the refund address `if trade.wallets.fromAsset.ID == waID`. The refund address always comes from the `fromWallet`. 

If my thinking is right, then answering this question should help. In what situation would we have no refund address = `len(match.MetaData.Proof.Script) == 0`?

Reply:
I think `no refund address = len(match.MetaData.Proof.Script) == 0` itself it always correct? But it's true from order status epoch up until match status `MakerSwapCast` for the maker and `TakerSwapCast` for the taker?

Reply:
Looks like `maybeDifferentWallet` resolves my concerns. Will retest and approve.